test/integration/targets/azure_rm_mysqlserver/tasks/main.yml

```diff
@@ -276,3 +297,14 @@
   assert:
     that:
       - output.changed == false
+
+- name: Delete second instance of MySQL Server
+  azure_rm_mysqlserver:
+    resource_group: "{{ resource_group }}second"
+    name: mysqlsrv{{ rpfx }}
+    state: absent
+  register: output
+- name: Assert the state has changed
```

Comment:
you don't need to wait till deletion done, using async mode to reduce test execution